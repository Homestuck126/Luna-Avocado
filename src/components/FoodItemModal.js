import React, { useState, useEffect } from "react";
import { View, Text, StyleSheet, Pressable, Modal } from "react-native";
import FoodListItem from "./FoodListItem";

const FoodItemModal = ({ isVisible, onClose, foodData }) => {
  const [modalVisible, setModalVisible] = useState(false);
  useEffect(() => {
    setModalVisible(isVisible);
  }, [isVisible]);
  const closeModal = () => {
    setModalVisible(false);
    onClose(false);
  };
  return (
    <Modal
      animationType="slide"
      transparent={true}
      visible={modalVisible}
      onRequestClose={() => {
        closeModal;
      }}
    >
      <View style={styles.centeredView}>
        <View style={styles.modalView}>
          <FoodListItem data={foodData} isPressable={false}></FoodListItem>
          <View style={styles.buttonContainer}>
            <Pressable
              style={({ pressed }) => [
                styles.button,
                styles.buttonClose,
                {
                  backgroundColor: pressed
                    ? "rgba(52,152,219,0.7)"
                    : "rgb(52,152,219)",
                },
              ]}
            >
              <Text style={styles.modalText}>Edit</Text>
            </Pressable>
            <Pressable
              style={({ pressed }) => [
                styles.button,
                styles.buttonClose,
                {
                  backgroundColor: pressed
                    ? "rgba(52,152,219,0.7)"
                    : "rgb(52,152,219)",
                },
              ]}
              onPress={() => {
                closeModal();
              }}
              testID="FoodItemModal:ClosePressable:Press"
            >
              <Text style={styles.modalText}>Close</Text>
            </Pressable>
          </View>
        </View>
      </View>
    </Modal>
  );
};

const styles = StyleSheet.create({
  centeredView: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    marginTop: 22,
  },
  modalView: {
    height: "24%",
    width: "80%",
    margin: 20,
    backgroundColor: "white",
    borderRadius: 20,
    borderWidth: 3,
    borderColor: "#708090",
    padding: 5,
    alignItems: "center",
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
  },
  modalText: {},
  button: {
    padding: 10,
    borderRadius: 8,
    width: "45%",
    alignItems: "center",
    marginHorizontal: 5,
  },
  buttonContainer: {
    justifyContent: "center",
    flexDirection: "row",
  },
  inputContainer: {
    flexDirection: "row",
    width: "100%",
    alignItems: "center",
    justifyContent: "space-between",
    backgroundColor: "rgb(245,245,245)",
    borderRadius: 8,
    paddingHorizontal: 14,
    marginBottom: 10,
  },
  input: {
    height: 40,
    width: 200,
    color: "#333",
    paddingVertical: 10,
    paddingRight: 10,
    fontSize: 16,
  },
});

export default FoodItemModal;
