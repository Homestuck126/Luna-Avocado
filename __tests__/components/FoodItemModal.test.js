import React from "react";
import { render, screen, fireEvent } from "@testing-library/react-native";
import FoodItemModal from "../../src/components/FoodItemModal";

describe("Food Item Modal", () => {
  it("calls onClose prop when Close pressable is pressed", () => {
    const mockOnPress = jest.fn();

    const foodData = {
      id: null,
      name: "testFood0",
      macros: [0, 0, 0, 0],
    };

    const { getByTestId } = render(
      <FoodItemModal
        isVisible={true}
        onClose={mockOnPress}
        foodData={foodData}
      ></FoodItemModal>
    );
    const closeButton = getByTestId("FoodItemModal:ClosePressable:Press");
    fireEvent.press(closeButton);

    expect(mockOnPress).toHaveBeenCalled();
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
