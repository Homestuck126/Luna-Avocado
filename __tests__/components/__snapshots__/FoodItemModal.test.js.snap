// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Food Item Modal calls onClose prop when Close pressable is pressed 1`] = `null`;

exports[`Food Item Modal renders modal with given data 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginTop": 22,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderColor": "#708090",
          "borderRadius": 20,
          "borderWidth": 3,
          "elevation": 5,
          "height": "24%",
          "margin": 20,
          "padding": 5,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 4,
          "width": "80%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "backgroundColor": "rgb(248,248,255)",
            },
            {
              "alignItems": "flex-start",
              "borderColor": "#C0C0C0",
              "borderRadius": 10,
              "borderWidth": 3,
              "height": 120,
              "margin": 10,
              "padding": 10,
              "width": "95%",
            },
          ]
        }
      >
        <Text
          style={
            {
              "fontSize": 18,
              "fontWeight": "bold",
              "marginBottom": 5,
            }
          }
        >
          testFood1
        </Text>
        <Text
          style={
            {
              "color": "#666",
              "fontSize": 14,
            }
          }
        >
          Calories: 
          100
        </Text>
        <Text
          style={
            {
              "color": "#666",
              "fontSize": 14,
            }
          }
        >
          Protein: 
          10
        </Text>
        <Text
          style={
            {
              "color": "#666",
              "fontSize": 14,
            }
          }
        >
          Carbs: 
          10
        </Text>
        <Text
          style={
            {
              "color": "#666",
              "fontSize": 14,
            }
          }
        >
          Fat: 
          10
        </Text>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "alignItems": "center",
                "borderRadius": 8,
                "marginHorizontal": 5,
                "padding": 10,
                "width": "45%",
              },
              undefined,
              {
                "backgroundColor": "rgb(52,152,219)",
              },
            ]
          }
        >
          <Text
            style={{}}
          >
            Edit
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "alignItems": "center",
                "borderRadius": 8,
                "marginHorizontal": 5,
                "padding": 10,
                "width": "45%",
              },
              undefined,
              {
                "backgroundColor": "rgb(52,152,219)",
              },
            ]
          }
          testID="FoodItemModal:ClosePressable:Press"
        >
          <Text
            style={{}}
          >
            Close
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
